Source/Diagnostics/ReducedDiags/MultiReducedDiags.cpp

```diff
@@ -35,75 +39,34 @@ MultiReducedDiags::MultiReducedDiags ()
     // if names are not given, reduced diags will not be done
     if ( m_plot_rd == 0 ) { return; }
 
-    // resize
-    m_multi_rd.resize(m_rd_names.size());
-
+    using CS = const std::string& ;
+    const auto reduced_diags_dictionary =
+        std::map<std::string, std::function<std::unique_ptr<ReducedDiags>(CS)>>{
+            {"ParticleEnergy",        [](CS s){return std::make_unique<ParticleEnergy>(s);}},
+            {"FieldEnergy",           [](CS s){return std::make_unique<FieldEnergy>(s);}},
+            {"FieldMaximum",          [](CS s){return std::make_unique<FieldMaximum>(s);}},
+            {"RhoMaximum",            [](CS s){return std::make_unique<RhoMaximum>(s);}},
+            {"BeamRelevant",          [](CS s){return std::make_unique<BeamRelevant>(s);}},
+            {"LoadBalanceCosts",      [](CS s){return std::make_unique<LoadBalanceCosts>(s);}},
+            {"LoadBalanceEfficiency", [](CS s){return std::make_unique<LoadBalanceEfficiency>(s);}},
+            {"ParticleHistogram",     [](CS s){return std::make_unique<ParticleHistogram>(s);}},
+            {"ParticleNumber",        [](CS s){return std::make_unique<ParticleNumber>(s);}},
+            {"ParticleExtrema",       [](CS s){return std::make_unique<ParticleExtrema>(s);}}
+        };
     // loop over all reduced diags
```

Comment:
```suggestion
    // loop over all reduced diags and fill m_multi_rd with requested reduced diags
```
I guess making this slightly more explicit could be useful for those of us not too familiar with the STL. :)